Source/WebCore/platform/graphics/PathSegmentData.cpp

```diff
@@ -0,0 +1,564 @@
+        boundingRect.extend(circleRect.minXMinYCorner());
+        boundingRect.extend(circleRect.maxXMaxYCorner());
+        return;
+    }
+
+    float x1 = center.x() + radius * cos(startAngle);
+    float y1 = center.y() + radius * sin(startAngle);
+
+    float x2 = center.x() + radius * cos(endAngle);
+    float y2 = center.y() + radius * sin(endAngle);
+
+    float startAngle = this->startAngle;
+    float endAngle = this->endAngle;
+
+    if (x1 > x2)
+        std::swap(x1, x2);
+
+    if (y1 > y2)
+        std::swap(y1, y2);
+
+    if (direction == RotationDirection::Counterclockwise) {
+        std::swap(startAngle, endAngle);
+        startAngle = angleInClockwise(startAngle, direction);
+    }
+
+    if (isInRange(float(0), startAngle, endAngle))
+        x2 = circleRect.maxX();
+
+    if (isInRange(angleInClockwise(piFloat / 2, direction), startAngle, endAngle))
+        y2 = circleRect.maxY();
+
+    if (isInRange(angleInClockwise(piFloat, direction), startAngle, endAngle))
+        x1 = circleRect.x();
+
+    if (isInRange(angleInClockwise(3 * piFloat / 2, direction), startAngle, endAngle))
+        y1 = circleRect.y();
+
+    boundingRect.extend({ x1, y1 });
+    boundingRect.extend({ x2, y2 });
+}
+
+void PathArc::addToImpl(PathImpl& impl) const
+{
+    impl.addArc(center, radius, startAngle, endAngle, direction);
+}
+
+WTF::TextStream& operator<<(WTF::TextStream& ts, const PathArc& data)
+{
+    ts << "add arc " << data.center << " " << data.radius  << " " << data.startAngle  << " " << data.endAngle  << " " << data.direction;
+    return ts;
+}
+
+FloatPoint PathEllipse::calculateEndPoint(const FloatPoint&, FloatPoint& lastMoveToPoint) const
+{
+    lastMoveToPoint = center + FloatSize { radiusX * cos(startAngle), - radiusY * sin(startAngle) };
+    auto endPoint = center + FloatSize { radiusX * cos(endAngle), - radiusY * sin(endAngle) };
+    if (!rotation)
+        return endPoint;
+
+    auto rotation = AffineTransform::makeRotation(deg2rad(this->rotation));
+    lastMoveToPoint = rotation.mapPoint(lastMoveToPoint);
+    return rotation.mapPoint(endPoint);
+}
+
+void PathEllipse::extendFastBoundingRect(const FloatPoint&, FloatRect& boundingRect) const
+{
+    auto minXMinYCorner = center - FloatSize { radiusX, radiusY };
+    auto maxXMaxYCorner = center + FloatSize { radiusX, radiusY };
+
+    if (!rotation) {
+        boundingRect.extend(minXMinYCorner);
+        boundingRect.extend(maxXMaxYCorner);
+        return;
+
+    }
+
+    auto rect = FloatRect { minXMinYCorner, maxXMaxYCorner };
+    auto rotation = AffineTransform::makeRotation(deg2rad(this->rotation));
+    rect = rotation.mapRect(rect);
+
+    boundingRect.extend(rect.minXMinYCorner());
+    boundingRect.extend(rect.maxXMaxYCorner());
+}
+
+void PathEllipse::extendBoundingRect(const FloatPoint& currentPoint, FloatRect& boundingRect) const
+{
+    // FIXME: Implement this function.
+    boundingRect.extend(currentPoint);
```

Comment:
What's the consequence of not implementing this for now? Could it mean that e.g. some SVG <ellipse> will return the wrong value for `getBBox()`? If this is user visible then I don't think we can avoid implementing it now.